tests/components/tfa_me/test_coordinator.py

```diff
@@ -0,0 +1,356 @@
+            CONF_IP_ADDRESS: "127.0.0.1",
+            CONF_NAME_WITH_STATION_ID: False,
+        },
+        unique_id="test-1234",
+    )
+    entry.add_to_hass(hass)
+    return entry
+
+
+@pytest.mark.asyncio
+async def test_update_data_with_ip(hass: HomeAssistant, tfa_me_mock_entry) -> None:
+    """Test normal update (with IP) with some sensor types."""
+    now = datetime.now().timestamp()
+
+    # Create dummy JSON reply
+    dummy_json = {
+        "gateway_id": "017654321",
+        "sensors": [
+            {
+                "sensor_id": "a21234567",
+                "name": "A21234567",
+                "timestamp": "2025-09-04T12:21:41Z",
+                "ts": int(now),
+                "measurements": {
+                    "rssi": {"value": "221", "unit": "/255"},
+                    "lowbatt": {"value": "0", "unit": "No"},
+                    "wind_direction": {"value": "8", "unit": ""},
+                    "wind_speed": {"value": "0.0", "unit": "m/s"},
+                    "wind_gust": {"value": "0.0", "unit": "m/s"},
+                },
+            },
+            {
+                "sensor_id": "a12345678",
+                "name": "A12345678",
+                "timestamp": "2025-09-05T06:46:31Z",
+                "ts": int(now),
+                "measurements": {
+                    "rssi": {"value": "216", "unit": "/255"},
+                    "lowbatt": {"value": "0", "unit": "No"},
+                    "rain": {"value": "29.2", "unit": "mm"},
+                },
+            },
+        ],
+    }
+
+    coordinator = TFAmeDataCoordinator(
+        hass,
+        tfa_me_mock_entry,
+        "127.0.0.1",
+        timedelta(seconds=30),
+        name_with_station_id=True,
+    )
+    coordinator.first_init = 1
+
+    # Patch TFAmeClient delivers JSON directly
+    with patch(
+        "homeassistant.components.tfa_me.coordinator.TFAmeClient",
+        autospec=True,
+    ) as mock_client_cls:
+        mock_client = AsyncMock()
+        mock_client.async_get_sensors.return_value = dummy_json
+        mock_client_cls.return_value = mock_client
+
+        # Request data
+        result = await coordinator._async_update_data()
+
+    # Asserts wind sensor
+    assert "sensor.017654321_a21234567_wind_direction" in result
+    assert "sensor.017654321_a21234567_wind_direction_txt" in result
+    assert "sensor.017654321_a21234567_wind_direction_deg" in result
+    assert "sensor.017654321_a21234567_wind_speed" in result
+    assert "sensor.017654321_a21234567_wind_gust" in result
+    assert "sensor.017654321_a21234567_rssi" in result
+    assert "sensor.017654321_a21234567_lowbatt" in result
+    assert "sensor.017654321_a21234567_lowbatt_txt" in result
+
+    # Asserts rain sensor
+    assert "sensor.017654321_a12345678_rain" in result
+    assert "sensor.017654321_a12345678_rssi" in result
+    assert "sensor.017654321_a12345678_lowbatt" in result
+    assert "sensor.017654321_a12345678_lowbatt_txt" in result
```

Comment:
We can use snapshot testing instead for the entities, but for just the coordinator, this should not be necessary.